src/coinselection/greedy.ts

```
@@ -32,6 +33,8 @@ function greedyCoinSelection(
   changeAddressGetter: ChangeAddressFromAssetGetter,
   sortFn: CompareUtxoFn
 ): CoinSelectionResult {
+  unspents = unspents.filter(utxo => !isBlindedUtxo(utxo));
```

can you explain this?

---

it avoid using blindedUtxos because the algorithm only works with unconfidential unspents

---

shouldn't it work also with conf utxos?

---

well we assume we got them already unblinded before coin selection

---

yeah, I got it now.. maybe better to raise an error here instead of filtering tho? It's mandatory that the unspents provided here have been unblinded eventually.. what do you think?

---

nACK. I think here a blinded utxo mens, no utxos.